### PR TITLE
roachtest: add more exceptions to bank/zerosum-restart

### LINKED ISSUE #33683: roachtest: acceptance/bank/zerosum-restart failed [skipped]
SHA: https://github.com/cockroachdb/cockroach/commits/19860c9f7b02a698109ff3905b93c384cdd8d6d5

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=acceptance/bank/zerosum-restart PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1090188&tab=buildLog

```
The test failed on master:
	test.go:696,bank.go:375,bank.go:569,acceptance.go:72: pq: cannot up-replicate to s3; missing gossiped StoreDescriptor
	test.go:696,cluster.go:1130,bank.go:247: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod stop teamcity-1090188-acceptance:1 returned:
		stderr:
		
		stdout:
		teamcity-1090188-acceptance: stopping and waiting: signal: killed

```

### PR DESCRIPTION
**Pull request by tbg**
Fixes #33683.

Release note: None

### CODE REVIEW

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33707)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/bank.go, line 348 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/33707#-LWBek475tt6BuF0rvcg:-LWBek475tt6BuF0rvch:b-3kso45) ([raw file](https://github.com/cockroachdb/cockroach/blob/29f70ccb9eb3b14d633d8e89d082a83ec403791b/pkg/cmd/roachtest/bank.go#L348)):*
> ```Go
> 
> func isExpectedRelocateError(err error) bool {
> 	return testutils.IsError(err, "(descriptor changed|unable to remove replica .* which is not present|unable to add replica .* which is already present|received invalid ChangeReplicasTrigger .* to remove self|breaker open|cannot up-replicate|error looking up store)")
> ```

This is getting a bit excessive. When I originally added these, there were explanations for why the various errors were occurring. Do you know why/how `error looking up store` is occurring? That one seems particularly suspicious.

**Review by andreimatei — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33707)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/bank.go, line 348 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/33707#-LWBek475tt6BuF0rvcg:-LWCkNAa4D2sdeJWDUhD:b-x4pz4t) ([raw file](https://github.com/cockroachdb/cockroach/blob/29f70ccb9eb3b14d633d8e89d082a83ec403791b/pkg/cmd/roachtest/bank.go#L348)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

This is getting a bit excessive. When I originally added these, there were explanations for why the various errors were occurring. Do you know why/how `error looking up store` is occurring? That one seems particularly suspicious. 
</blockquote></details>

yeah... Is this the new `testing.RetriableErr()` of ill repute?

**Comment by tbg** [comment]
They're not suspicious if you ask me:

https://github.com/cockroachdb/cockroach/blob/2558dccbcda09c0eae28f816a05adb8620155175/pkg/sql/relocate.go#L171-L183

https://github.com/cockroachdb/cockroach/blob/bd00a26afd75b84160735ad848a7d48e3cc8a660/pkg/storage/replica_command.go#L1232-L1239

I agree that it's unsavory. These errors can also happen to users. Ultimately they're the result on relying on an asynchronous mechanism (Gossip). To adress them, I think, we ultimately need to either a) poll Gossip for a bit before declaring that something "isn't there" or b) have fallbacks to the KV store for everything (again with a timeout, to make this more reliable).

I won't be working on that anytime soon though, so my question is whether you prefer this to merge as-is (I'm happy to backfill the error descriptions with code snippets) or whether we'll want to skip the test again.

**Review by petermattis — COMMENTED** [review]
> I won't be working on that anytime soon though, so my question is whether you prefer this to merge as-is (I'm happy to backfill the error descriptions with code snippets) or whether we'll want to skip the test again.

I'd prefer to leave the test as skipped until someone has time to more thoroughly investigate and fix. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33707#-:-LWIGcmi-3lhTcm8h6Ag:b-a8or2c)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by tbg** [comment]
> I'd prefer to leave the test as skipped until someone has time to more thoroughly investigate and fix.

👍  done, changed the PR accordingly and also referenced the known failure modes I know of.

bors r=petermattis

**Comment by tbg** [comment]
Had removed a bracket too many.

**Comment by tbg** [comment]
bors r=petermattis

**Comment by tbg** [comment]
TestLint flake.

bors r=petermattis

File: pkg/cmd/roachtest/acceptance.go [hunk 1]
```diff
@@ -33,7 +37,7 @@ func registerAcceptance(r *registry) {
 		{"bank/cluster-recovery", runBankClusterRecovery},
 		{"bank/node-restart", runBankNodeRestart},
 		{"bank/zerosum-splits", runBankNodeZeroSum},
-		{"bank/zerosum-restart", runBankZeroSumRestart},
+		// {"bank/zerosum-restart", runBankZeroSumRestart},
```
Comment (andreimatei) [inline pkg/cmd/roachtest/acceptance.go]: wait - why comment this out instead of putting a `t.Skip()` at the beginning of `runBankZeroSumRestart()` ?

**Pull request merged**